k8s/quickstart/server-statefulset.yaml

```diff
@@ -42,6 +42,14 @@ spec:
             initialDelaySeconds: 15
             periodSeconds: 60
             timeoutSeconds: 3
+          readinessProbe:
```

Comment:
could you prove it works, and what happens with another tutorials? I saw another tutos dont have readiness probe, so, may we add it to all of them? (as a note, for now there is no difference between normal and `--shallow`, but it is the idea to implement it in a future, so it is good idea to set it)

Reply:
Hi Marcos, I tested the behavior of adding that bit by placing a dummy sleep on the SPIRE server, and it works perfectly (server not transitioning into ready status until it is really ready to serve a request). About the other tutorials, all of them (except _OIDC_ ones) are built in top of this quickstart, so basically adding this _readinessProbe_ to the SPIRE server is adding it to all the tutorials. I'll add this probe to the _OIDC_ tutorials as well.

Reply:
I think your test is reasonable, as long as it differentiates between liveness and readiness probes effectively enough. The sleep makes sense for initial readiness, and otherwise since both probes do the exact same check it might be tricky - the only real difference is that the readiness check is more aggressive.

It probably is worth adding to the other tutorials as well, although again both do the exact same check so all that will happen is that a service will fall out of readiness for a few seconds before being terminated once the liveness check fails. It's worth adding to the tutorials as a best practice/starting point.

Once --shallow is implemented then I *think* differentiation is a bit clearer, so it make sense to me that liveness probes use the default, and readiness probes use --shallow.